eclass/yarn.eclass

```diff
@@ -0,0 +1,154 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: yarn.eclass
+# @MAINTAINER:
+# Andrew Udvare <audvare@gmail.com>
+# @AUTHOR:
+# Andrew Udvare <audvare@gmail.com>
+# @BLURB: Install a Node-based package offline with Yarn.
+# @DESCRIPTION:
+
+case ${EAPI:-0} in
+	8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} unsupported." ;;
+esac
+
+inherit edo multiprocessing
+
+EXPORT_FUNCTIONS src_unpack src_prepare src_configure src_compile src_install
+
+if [[ -z ${_YARN_ECLASS} ]]; then
+	# @ECLASS_VARIABLE: YARN_PKGS
+	# @DESCRIPTION:
+	# Bash array of Yarn package specifications in format [@SCOPE/]NAME-VERSION.
+	# @CODE
+	#
+	# YARN_PKGS=( @types/type-pkg-1.0.0 )
+	#
+	# @CODE
+
+	# @ECLASS_VARIABLE: _YARN_DISTFILES
+	# @_INTERNAL:
+	# @DESCRIPTION:
+	# Array of distfile basenames. The output path for a tarball may be
+	# different than the basename taken off the URI.
+	_YARN_DISTFILES=()
+
+	# @FUNCTION: yarn_set_globals
+	# @DESCRIPTION:
+	# This must be called after defining YARN_PKGS in global scope. This
+	# function sets up YARN_SRC_URI which must be added to SRC_URI. If you need
+	# to be certain that BDEPEND/RDEPEND/RESTRICT/SLOT is empty, set those
+	# after calling this.
+	yarn_set_globals() {
+		# shellcheck disable=SC2034
```

Comment:
I'd personally prefer we not add the shellcheck annotations given they feel a bit noisy, but I can understand why you have.